### PR TITLE
Move the non-legacy code to src

### PR DESCRIPTION
**Pull request by laryjulien**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       | #4548
| License       | MIT
| Doc PR        | To be completed

### CODE REVIEW

File: src/Bridge/Doctrine/Orm/State/UriVariablesHandlerTrait.php [hunk 1]
```diff
@@ -38,35 +38,19 @@ private function handleUriVariables(QueryBuilder $queryBuilder, array $identifie
...
+                $expression = $queryBuilder->expr()->eq("{$joinAlias}.{$propertyIdentifier}", $placeholder);
             } elseif ($property = $uriVariable->getProperty()) {
                 $propertyIdentifier = $uriVariable->getIdentifiers()[0];
                 $joinAlias = $queryNameGenerator->generateJoinAlias($property);
 
-                $queryBuilder->join(
-                    "$alias.$property",
-                    $joinAlias,
-                );
+                $queryBuilder->join("$alias.$property", $joinAlias, );
```
Comment (kaznovac) [inline src/Bridge/Doctrine/Orm/State/UriVariablesHandlerTrait.php:48]: extra comma (probably all these formatting changes should be reverted) 

```suggestion
                $queryBuilder->join("$alias.$property", $joinAlias);
```

**Review by kaznovac — COMMENTED** [review]
this pr should address file move and namespace changes.

the formatting changes (if necessary) should be suggested as part of a separate pr.

File: src/Bridge/Doctrine/Orm/State/UriVariablesHandlerTrait.php (hunk 1 above)
Comment (laryjulien) [reply src/Bridge/Doctrine/Orm/State/UriVariablesHandlerTrait.php]: As there are not intended for this PR, I have reverted all code style changes.
Thank you.

**Comment by laryjulien** [comment]
FQCN (fully qualified class name) are **not** updated in CHANGELOG.md

**Comment by soyuka** [comment]
Huge work thank you very much @laryjulien @kevin-verschaeve @GrimpEreau @FloVnst !

**Pull request merged**